internal/sidecarlogresults/sidecarlogresults_test.go

```diff
@@ -0,0 +1,148 @@
+package sidecarlogresults
+
+import (
+	"bytes"
+	"encoding/json"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/tektoncd/pipeline/test/diff"
+)
+
+func TestLookForResults_FanOutAndWait(t *testing.T) {
+	for _, c := range []struct {
+		desc    string
+		results []SidecarLogResult
+	}{{
+		desc: "multiple results",
+		results: []SidecarLogResult{{
+			Name:  "foo",
+			Value: "bar",
+		}, {
+			Name:  "foo2",
+			Value: "bar2",
+		}},
+	}} {
+		t.Run(c.desc, func(t *testing.T) {
+			dir := t.TempDir()
+			resultNames := []string{}
+			wantResults := []string{}
+			for _, result := range c.results {
+				createResult(t, dir, result.Name, result.Value)
+				resultNames = append(resultNames, result.Name)
+				encodedResult, err := json.Marshal(result)
+				if err != nil {
+					t.Error(err)
+				}
+				// encode adds a newline character at the end.
+				// We need to do the same before comparing
+				encodedResult = append(encodedResult, '\n')
+				wantResults = append(wantResults, string(encodedResult))
+			}
+			dir2 := t.TempDir()
+			go createRun(t, dir2, false, true)
+			got := new(bytes.Buffer)
+			err := LookForResults(got, dir2, dir, strings.Join(resultNames, ","))
+			if err != nil {
+				t.Fatalf("Did not expect any error but got: %v", err)
+			}
+			for _, res := range wantResults {
+				if !strings.Contains(got.String(), res) {
+					t.Errorf("Expected result %s to be in %s", res, got.String())
+				}
+			}
```

Comment:
prefer sorting the results and using cmp.Diff

Reply:
sure